simulationdataschema/atoms_state.py

```diff
@@ -0,0 +1,571 @@
+        """,
+        a_eln=ELNAnnotation(component="StringEditQuantity"),
+    )
+
+    def resolve_u_interactions(self, logger: BoundLogger) -> Optional[tuple]:
+        """
+        Resolves the Hubbard interactions (u, up, jh) from the Slater integrals (F0, F2, F4).
+
+        Args:
+            logger (BoundLogger): The logger to log messages.
+
+        Returns:
+            (Optional[tuple]): The Hubbard interactions (u, up, jh).
+        """
+        if self.slater_integrals is None or len(self.slater_integrals) == 3:
+            logger.warning(
+                "Could not find `slater_integrals` or the length is not three."
+            )
+            return None
+        f0 = self.slater_integrals[0]
+        f2 = self.slater_integrals[1]
+        f4 = self.slater_integrals[2]
+        u_interaction = (
+            ((2.0 / 7.0) ** 2)
+            * (f0 + 5.0 * f2 + 9.0 * f4)
+            / (4.0 * np.pi)
+            * ureg("joule")
+        )
+        up_interaction = (
+            ((2.0 / 7.0) ** 2)
+            * (f0 - 5.0 * f2 + 3.0 * f4 / 2.0)
+            / (4.0 * np.pi)
+            * ureg("joule")
+        )
+        jh_interaction = (
+            ((2.0 / 7.0) ** 2)
+            * (5.0 * f2 + 15.0 * f4 / 4.0)
+            / (4.0 * np.pi)
+            * ureg("joule")
+        )
+        return u_interaction, up_interaction, jh_interaction
+
+    def resolve_u_effective(self, logger: BoundLogger) -> Optional[np.float64]:
+        """
+        Resolves the effective U parameter (u - j).
+
+        Args:
+            logger (BoundLogger): The logger to log messages.
+
+        Returns:
+            (Optional[np.float64]): The effective U parameter.
+        """
+        if self.u is None or self.j is None:
+            logger.warning(
+                "Could not find `HubbardInteractions.u` or `HubbardInteractions.j`."
+            )
+            return None
+        return self.u - self.j
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+        # Obtain (u, up, jh) from slater_integrals
+        if self.u is None and self.up is None and self.jh is None:
+            self.u, self.up, self.jh = self.resolve_u_interactions(logger)
+
+        # If u_effective is not available, calculate it
+        if self.u_effective is None:
+            self.u_effective = self.resolve_u_effective(logger)
+
+        # Check if length of `orbitals_ref` is the same as the length of `umn`:
+        if self.umn is not None and self.orbitals_ref is not None:
+            if len(self.umn) != len(self.orbitals_ref):
+                logger.error(
+                    "The length of `HubbardInteractions.umn` does not coincide with length of `HubbardInteractions.orbitals_ref`."
+                )
+
+
+class AtomsState(ArchiveSection):
+    """
+    A base section to define each atom state information.
+    """
+
+    # ? constraint to the normal chemical elements (no 'X' as defined in ASE included)
```

Comment:
Was this a remark from me or you?

Reply:
Be careful: this might result in breaking changes with the parsers.

Luca Ghiringhelli had some calculations that were using ghost atoms (`X`).
Maybe you should ask him how we can accommodate him.

Reply:
Ok... I can add them back, and add in the description that  `"... 'X' is used in the context of ghots atoms, see e.g., <href>"`.

Reply:
I just wanted to separate a bit the real elements from other kind of concepts

Reply:
I think that's more in line with its current use, yes.

Reply:
> I just wanted to separate a bit the real elements from other concepts, perhaps for these models it is easier to define labels as a list, or not defining them at all.

Yeah, they're a pain, since they also prevent the computation of physical quantities like mass density...
Idk, I don't like them either, but they are used by times...

Reply:
I will keep them out for the time being. I have the feeling that I need to see an use-case before we make any decision. I'll add a comment.